src/ewatercycle/analysis/__init__.py

```diff
@@ -2,11 +2,26 @@
 from typing import Tuple, Union
 
 import matplotlib.pyplot as plt
+import numpy as np
 import pandas as pd
 from hydrostats import metrics
 from matplotlib.dates import DateFormatter
 
 
+def downsample(df, nrows=100, agg="mean"):
```

Comment:
This is a utility method that I feel should not be part of the public analysis API. Can it be renamed to `_downsample()` or moved to util.py .

If you want to keep it public then the arguments should have a type and docstring. The docstring should mention when aggregation is performed.